sdk/storage/azure-storage-datamovement/src/filesystem.cpp

```diff
@@ -327,6 +339,20 @@ namespace Azure { namespace Storage { namespace _internal {
     }
   }
 
+  void AzurePathToLocalpath(std::string& azurePath)
+  {}
+
+  std::string GetParentDir(const std::string& blobPath)
+  {
+    auto pos = blobPath.find_last_of('\\');
+    if (-1 != pos)
+    {
+      return blobPath.substr(0, pos);
+    }
+
+    return "";
```

Comment:
I got what you want to do when I was looking at the code where this function is called.

If we add functions in this header, the function should be generic enough, because not only we but also other developers in the future will look at this header and use functions in it.

In this specific case, returning empty is ok if `blobPath` is already tail of a path.
Other than that, I think we should also make this function able to work when `blobPath` is absolute.

Reply:
Moved it to a internal place.

Reply:
This is to get its relative parent path. If it's only a name without directory, the function should return an empty directory path. We'd use this function to get relative directory, and append to destination root to check whether the require directory exists.

Reply:
> This is to get its relative parent path. If it's only a name without directory, the function should return an empty directory path. We'd use this function to get relative directory, and append to destination root to check whether the require directory exists.

Yes, I understand. My concern is that it cannot be inferred from the function signature (function name, parameter name) that this function only works for relative path. What if someday some people find this function and use it for absolute path, only find out it doesn't work. That won't be good.